client/src/components/Workbench/Workbench.js

```diff
@@ -76,15 +78,32 @@ function Workbench() {
                 <Route path={`${path}${ROUTES.DATA}`}>
                   <DataFiles />
                 </Route>
-                <Route
-                  path={`${path}${ROUTES.APPLICATIONS}`}
-                  component={Applications}
-                />
+                {showApps ? (
+                  <Route
+                    path={`${path}${ROUTES.APPLICATIONS}`}
+                    component={Applications}
+                  />
+                ) : (
+                  <Redirect
+                    from={`${path}${ROUTES.APPLICATIONS}`}
+                    to={`${path}${ROUTES.DASHBOARD}`}
+                  />
+                )}
                 <Route
                   path={`${path}${ROUTES.ALLOCATIONS}`}
                   component={Allocations}
                 />
-                <Route path={`${path}${ROUTES.HISTORY}`} component={History} />
+                {showApps ? (
+                  <Route
+                    path={`${path}${ROUTES.HISTORY}`}
+                    component={History}
+                  />
+                ) : (
+                  <Redirect
+                    from={`${path}${ROUTES.HISTORY}`}
+                    to={`${path}${ROUTES.DASHBOARD}`}
+                  />
```

Comment:
👍 This seems sensible.

Reply:
There is a catch-all redirect on line 120 so these redirects are unneeded.

------------

but now or even after you removed them, there is also an issue as if you refresh on `https://cep.dev/workbench/applications` you get redirected to dashboard even if `"showApps": True` in your settings.

It looks like due to `client/src/redux/reducers/workbench.reducers.js` and the `loading` starting as false, the routes here are being used as the `loading` in line 65 is initially false --> which leads to a state where `loading` is false and `showApps` is `undefined`.

Reply:
Ah. Thank you.